conf/imx_hab4/README.txt

```diff
@@ -0,0 +1,268 @@
+=> fuse prog 5 3 0x364727C3
+=> fuse prog 5 4 0x8DD52832
+=> fuse prog 5 5 0xF158F65F
+=> fuse prog 5 6 0xA71BBE78
+=> fuse prog 5 7 0xA3AD024A
+
+5. Build SPL with HAB support
+Build U-Boot: CONFIG_IMX_HAB
+
+When SPL is built:
+$ cat SPL.log | grep HAB
+HAB Blocks:   0x2f010400 0x00000000 0x00016c00
+
+6. Then sign the SPL image
+$ cp SPL /tmp/spl,sign
+$ cd /tmp/spl,sign
+
+Replace the commented out Blocks line at the bottom with the HAB blocks
+information from the previous step
+$ vi spl,sign/u-boot-spl-sign.csf-template
+
+Use the Code Signing Tool binary to generate the signature
+$ ./cst -o csf-spl.bin -i u-boot-spl-sign.csf-template
+
+Now append the signature to the SPL image
+$ cat SPL  csf-spl.bin > SPL.signed
+
+7. Install this SPL.signed image via SDP (uuu bootloader.uuu)
+
+8. Boot the image and check the HAB events for errors (no events is a pass!)
+=> hab_status
+Secure boot disabled
+HAB Configuration: 0xf0, HAB State: 0x66No HAB Events Found!
+
+9. Close the device by burning SEC_CONFIG fuse
+SEC_CONFIG[1] is on the imx7ulp is at bank29 word 6.
+The bit to be fused in order to enable secure boot is the lastone (BIT 31)
+
+=> fuse prog 29 6 0x80000000
+
+10. Reboot your board and check the HAB status
+
+=> hab_status
+Secure boot enabled
+HAB Configuration: 0xcc, HAB State: 0x99No HAB Events Found!
+
+Upgrades via SDP after the device was closed
+============================================
+Once the device has been closed, only signed images will be able to
+run on the processor.
+
+Support for upgrade via the Serial Download Protocol using the MFG tools 
+continues to be possible but with certain caveats:
+
+SDP will require that the CSF is modified to check the DCD table and that the 
+signed image DCD address is cleared in the IVT header (since the SDP protocol
+clears the DCD table address).
+
+To handle both requirements in Aeler we need to sign the SPL-mfg differently to 
+how we sign SPL just so it can boot, bring up the MFG kernel and
+manage the upgrade  as usual.
+
+In all cases HAB (high assurance boot) need to be enabled in the config.
+
+U-Boot on the MFG and the product release don't need to be built any 
+differently since it is not signed for HAB. 
+
+===============
+Signing SPL MFG 
+===============
+Use the following scripts to handle the DCD as previously described.
+Once SPLMfg has been built, check the DCD address by
+
+/tools/mkimage -l SPL
+
+Image Type:   Freescale IMX Boot Image
+Image Ver:    2 (i.MX53/6/7 compatible)
+Mode:         DCD
+Data Size:    147552 Bytes = 144.09 KiB = 0.14 MiB
+Load Address: 2f010420
+Entry Point:  2f011000
+HAB Blocks:   0x2f010400 0x00000000 0x00021c00
+DCD Blocks:   0x00910000 0x0000002c 0x00000258
+
+Notice that the DCD address is currently hardcoded to a wrong address
+by U-boot: on the imx7ulpea this address should be 0x2f010000.
+
+This information will be used to generate the Command Sequence File
+(CSF) used to sign the image.
+
+mod_4_mfgtool.sh
```

Comment:
We should have this script available separately, can be stored at the same folder as we have print_fuses.